src/core/error_handling.c

```diff
@@ -88,55 +149,55 @@ format_exception_traceback(PyObject* type, PyObject* value, PyObject* traceback)
   return result;
 }
 
+/**
+ * Wrap the exception in a Javascript PythonError object.
+ *
+ * The return value of this function is always a valid hiwire ID to an error
+ * object. It never returns NULL.
+ *
+ * We are cautious about leaking the Python stack frame, so we don't increment
+ * the reference count on the exception object, we just store a pointer to it.
+ * Later we can check if this pointer is equal to sys.last_value and if so
+ * restore the exception (see restore_sys_last_exception).
+ *
+ * WARNING: dereferencing the error pointer stored on the PythonError is a
+ * use-after-free bug.
+ */
 JsRef
-wrap_exception(bool attach_python_error)
+wrap_exception()
 {
   bool success = false;
-  PyObject* type = NULL;
-  PyObject* value = NULL;
-  PyObject* traceback = NULL;
+  PyObject* type;
+  PyObject* value;
+  PyObject* traceback;
```

Comment:
Wouldn't it be be better to explicitly initialize it to `NULL` as before, before passing it to functions?

Reply:
I suppose so. The functions unconditionally write to these by using them as arguments to [`PyErr_Fetch`](https://docs.python.org/3/c-api/exceptions.html?highlight=pyerr_fetch#c.PyErr_Fetch), but it is harmless to initialize to `NULL`.